L08-bytecodes/src/main/java/ru/shikhovtsev/proxy/IoCContainer.java

```diff
@@ -0,0 +1,51 @@
+package ru.shikhovtsev.proxy;
+
+import ru.shikhovtsev.Log;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class IoCContainer {
+
+  @SuppressWarnings("unchecked")
+  public <U, T extends U> U getProxiedObject(Class<U> castTo, T obj) {
+    if (!castTo.isInterface()) {
+      throw new RuntimeException("The first parameter must be an interface!");
+    }
+
+    if (Arrays.stream(obj.getClass().getMethods()).anyMatch(m -> m.isAnnotationPresent(Log.class))) {
```

Comment:
Вынести в метод. Для читаемости.
```suggestion
    if (hasAnnotation(...)) {
```